src/content/blog/2022-08-08-new-config-system-part-3.md

```diff
@@ -0,0 +1,144 @@
+---
+layout: post
+title: "ESLint's new config system, Part 3: Developer preview"
+teaser: "While the new config system isn't yet incorporated into the CLI, it is available via API for developers to test."
+tags:
+  - Configuration
+  - Command Line
+authors:
+  - nzakas
+categories:
+  - API Changes
+---
+
+In my [previous post](https://eslint.org/blog/2022/08/new-config-system-part-2/), I explained the fundamental concepts of using the new "flat" config system. The new config system isn't yet tied into the CLI while we do more internal testing, but we did want to give the ESLint community a chance to experiment with flat config while we work on incorporating it into the CLI. So ESLint v8.21.0 incorporates several ways to try out flat config as we work on it. Please keep in mind that everything mentioned in this post is experimental and we would love your feedback as you try it out.
+
+## Using flat config with the `Linter` class
+
+If you are currently using `Linter` from the `eslint` package, you can enable flat config by setting `configType: "flat"` as an option on the constructor. Here's an example:
+
+```js
+const linter = new Linter({ configType: "flat" });
+
+const messages = linter.verify("new Map()", {
+    languageOptions: {
+        ecmaVersion: 5,
+        sourceType: "script"
+    },
+    rules: {
+        "no-undef": "error"
+    }
+}, "filename.js");
+```
+
+When you pass `configType: "flat"` as an option, `Linter` expects that any configuration objects that are passed to `verify()` will be in flat config format. In this example, the second argument to `verify()` is a flat config object (you can pass either a single object or an array of objects). Any call to `verify()` will assume that the text being linted is a JavaScript file whose filename ends with `.js`, but it's always a good idea to pass in an explicit filename as the third argument.
+
+While this base case works the same regardless of which config system you're using, there are some important differences:
+
+* `defineRule()`, `defineRules()`, and `defineParser()` now throw errors. Runtime plugins (discussed in my previous post) make these methods obsolete.
+* `getRules()` also throws an error. This method would return different data depending on when it was called, so it can't be used with flat config.
+
+## Using flat config with the `ESLint` class
+
+While implementing flat config we discovered that it would be too difficult to create option to switch between config systems like we did for `Linter`. Instead, we created a `FlatESLint` class that encapsulates all of the existing functionality in `ESLint` but uses flat config instead of eslintrc. The `FlatESLint` class is intended only as a preview of functionality; once we switch over to flat config permanently, the current `ESLint` class will be deleted and `FlatESLint` will be renamed to `ESLint`.
```

Comment:
```suggestion
While implementing flat config we discovered that it would be too difficult to create an option to switch between config systems like we did for `Linter`. Instead, we created a `FlatESLint` class that encapsulates all of the existing functionality in `ESLint` but uses flat config instead of eslintrc. The `FlatESLint` class is intended only as a preview of functionality; once we switch over to flat config permanently, the current `ESLint` class will be deleted and `FlatESLint` will be renamed to `ESLint`.
```